feat: generate sql from idl

trankhacvy opened the pull request.

trankhacvy committed:
feat: generate sql from idl

Commit:
Added underscore lamports

AbhirupAbhi commented:
here are a few points to consider.

- One small change we have added, the lamports field has been renamed to _lamports to denote it is an internal field. Please pull the changes.
- Remove uncessary files - the workflow file is there, and we have not yet set up deployment
- no need for an .env with SHYFT API key and gaID, please remove
- A few more things can be displayed on the UI, like the number of accounts and custom types
- if possible, add a download button which creates and adds a download button, which downloads the schema as an .sql file
- From a design pov, the space between the header to IDL to SQL, and the space between the code block area should increase, footer should be visible on scroll (areas are marked by yellow)
- increase the width of the generate button so that there is space before and after the word generate. (marked by blue, that is approx)

Here are a few points by me:

- The title IDL to SQL is ok, we can add a small sub-line which explains the purpose of the project
- If possible please display the account discriminators in base64 encoded format, this is mostly used by internal team to test gPa calls
- generated sql section should have a heading
- the top github icon should point to this project in the repo, as the bottom one is already pointing to the shyft repo
- can add a readme on how to run the code, skipping the .env details
- please update the logo in the favicon, we no longer use the pink-purple theme

trankhacvy committed:
remove unuse code

trankhacvy commented:
> here are a few points to consider.
> 
> * [x] One small change we have added, the lamports field has been renamed to _lamports to denote it is an internal field. Please pull the changes.
> * [x] Remove uncessary files - the workflow file is there, and we have not yet set up deployment
> * [x] no need for an .env with SHYFT API key and gaID, please remove
> * A few more things can be displayed on the UI, like the number of accounts and custom types
> * if possible, add a download button which creates and adds a download button, which downloads the schema as an .sql file
> * [x] From a design pov, the space between the header to IDL to SQL, and the space between the code block area should increase, footer should be visible on scroll (areas are marked by yellow)
> * [x] increase the width of the generate button so that there is space before and after the word generate. (marked by blue, that is approx)
> 
> Here are a few points by me:
> 
> * [x] The title IDL to SQL is ok, we can add a small sub-line which explains the purpose of the project
> * If possible please display the account discriminators in base64 encoded format, this is mostly used by internal team to test gPa calls
> * generated sql section should have a heading
> * [x] the top github icon should point to this project in the repo, as the bottom one is already pointing to the shyft repo
> * [x] can add a readme on how to run the code, skipping the .env details
> * [x] please update the logo in the favicon, we no longer use the pink-purple theme

- The library I use doesn't support download but we already have copy button. I think it's still fine.
- Added account count.
- How and where we display "account discriminators"?

trankhacvy committed:
add download button

trankhacvy committed:
add graphql link
